src/vivarium_public_health/risks/effect.py

```diff
@@ -316,3 +320,158 @@ def _get_risk_exposure_class(self, builder: Builder) -> Risk:
+        this component.
+        """
+        return {
+            self.name: {
+                "data_sources": {
+                    "relative_risk": f"{self.risk}.relative_risk",
+                    "population_attributable_fraction": f"{self.risk}.population_attributable_fraction",
+                },
+            }
+        }
+
+    @property
+    def columns_created(self) -> List[str]:
+        return [f"{self.risk.name}_exposure"]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        exposure_col = pd.DataFrame(columns=self.columns_created)
+        self.population_view.update(exposure_col)
+
+    def on_time_step_prepare(self, event: Event) -> None:
+        # exposure_values = self.exposure(event.index)
+        # TODO: temporary workaround, investigate why exposure is always 0
+        exposure_values = np.linspace(3, 7, num=len(event.index))
+        exposure_col = pd.DataFrame(exposure_values, columns=self.columns_created).astype(
+            "object"
+        )
+        self.population_view.update(exposure_col)
+
+    #################
+    # Setup methods #
+    #################
+
+    def build_all_lookup_tables(self, builder: Builder) -> None:
+        rr_data = self.get_relative_risk_data(builder)
+        # TODO: add check that rr_data is parametrize by exposure
+        def define_rr_intervals(df: pd.DataFrame) -> pd.DataFrame:
+            max_exposure_row = df.tail(1).copy()
+            max_exposure_row["parameter"] = 9999
+            rr_data = pd.concat([df, max_exposure_row]).reset_index()
+            rr_data["left_exposure"] = [0] + rr_data["parameter"][:-1].tolist()
+            rr_data["left_rr"] = [rr_data["value"].min()] + rr_data["value"][:-1].tolist()
+            rr_data["right_exposure"] = rr_data["parameter"]
+            rr_data["right_rr"] = rr_data["value"]
+            return rr_data[
+                ["parameter", "left_exposure", "left_rr", "right_exposure", "right_rr"]
+            ]
+
+        # define exposure and rr interval columns
+        demographic_cols = [
+            col for col in rr_data.columns if col != "parameter" and col != "value"
+        ]
+        rr_data = (
+            rr_data.groupby(demographic_cols)
+            .apply(define_rr_intervals)
+            .reset_index(level=-1, drop=True)
+            .reset_index()
+        )
+        rr_data = rr_data.drop("parameter", axis=1)
+        rr_data[f"{self.risk.name}_exposure_start"] = rr_data["left_exposure"]
+        rr_data[f"{self.risk.name}_exposure_end"] = rr_data["right_exposure"]
+        # build lookup table
+        rr_value_cols = ["left_exposure", "left_rr", "right_exposure", "right_rr"]
+        self.lookup_tables["relative_risk"] = self.build_lookup_table(
+            builder, rr_data, rr_value_cols
+        )
+
+        # paf_data = self.get_filtered_data(
```

Comment:
This was done temporarily to avoid having to put PAF data in the artifact.

Reply:
You don't need data in the artifact though right? You can just configure it so that it loads it from that key.